### PR TITLE
Feat/deployment tool

### PR DESCRIPTION
**Pull request by adithya1012**
Added Deployment Tool 
Handled recursive call after the response is processed. 
keep a count of LLM mistakes and exit if it is stuck.

### CODE REVIEW

File: src/core/index.ts [hunk 1]
```diff
@@ -138,24 +153,41 @@ export class DevAssist {
...
+						// 	recussiveCall = false;
+						// }
+						if (block.name === "ask_followup_question" || block.name === "attempt_completion") {
+							recussiveCall = false;
 						}
+						if (block.name === "deploy_to_cloud") {
+							deployToolCalled = true;
+						}
+						await this.presentAssistantMessage(block);
+						// if (block["name"] === "ask_followup_question" || block["name"] === "attempt_completion") {
```
Comment (hdprajwal) [inline src/core/index.ts:188]: This goes into an infinite loop can you add the following code to fix it

```ts
						if (this.breakLoop) {
							this.breakLoop = false;
							recussiveCall = false;
							break;
						}
```

File: src/core/index.ts [hunk 2]
```diff
@@ -343,6 +391,102 @@ export class DevAssist {
...
+						} else {
+							this.informUser(
+								"Please install above dependencies and create a new chat with the same prompt to proceed \n\n"
+							);
+							this.userMessageContent.push({
+								type: "text",
+								text: `User did not install the dependencies. He infomred to install the requrement to host the application in cloud. Once you get the any response from user, you SHOULD call deploy_to_cloud tool again to check for dependencies.`,
+							});
+							this.deployToolCalled = false;
+							this.repoCreated = true; // Set to true to avoid the first check
```
Comment (hdprajwal) [inline src/core/index.ts:489]: set the breakloop to true to fix the infinite loop

Comment (hdprajwal) [reply src/core/index.ts:489]:
```ts
this.breakLoop = true;
```

**Review by hdprajwal — APPROVED** [review]
Fixes the issue of infinite loop.

**Review by nidhitalastha — APPROVED** [review]
LGTM.

**Review by Sinduja98 — APPROVED** [review]
Looks good.

**Review by jahnavib26 — APPROVED** [review]

**Pull request merged**